### PR TITLE
Use for...of syntax for SnippetExtractors.

### PR DESCRIPTION
**Pull request by clin1234**

### CODE REVIEW

File: out/utils/search.js [hunk 1]
```diff
@@ -21,21 +21,18 @@ async function search(keyword) {
...
-                const extractor = extractors_1.default[i];
+            for (const extractor of extractors_1.default) {
                 if (extractor.isEnabled()) {
                     const urls = await extractor.extractURLFromKeyword(keyword);
                     for (const y in urls) {
                         fetchResult = await fetchPageContent_1.fetchPageTextContent(urls[y]);
                         results = results.concat(extractor.extractSnippets(fetchResult));
                         vscode.window.setStatusBarMessage(`${extractor.name} (${y}/${urls.length}): ${results.length} results`, 2000);
-                        if (results.length >= config.settings.maxResults) {
+                        if (results.length >= config.settings.maxResults)
```
Comment (Kek5chen) [inline out/utils/search.js:31]: nice

**Comment by Kek5chen** [comment]
So far it looks good, commencing to testing...

**Comment by hieunc229** [comment]
There are few things:
1. The code casing, please use camelCase
2. You can save a fetch trip for each URL (can be done by adding a protected function to handle URL the ExtractorAbstract class, then override it in the IdeoneExtractor). Otherwise, fetching 12 URL will become 24 requests
3. (optional but recommend) Use the existing function to get textContent instead of using 'request'
4. For the GistExtractor, I used the question mark operator because sometimes, somehow the querySelector return undefined. It will throw Error if you use '.parentNode'

**Comment by Kek5chen** [comment]
Outdated pull. Please pull latest master first before editing, then reopen new pr.

**Pull request closed without merging**